### ISSUE TITLE
Express transaction names broken in v2.6.0+

### ISSUE THREAD

**Issue by MartinKolarik** [elastic/apm-agent-nodejs#1008]
**To Reproduce**
```js
require('elastic-apm-node').start();

const express = require('express');
const server = express();

// apm-agent-nodejs < 2.6.0: setting default transaction name: GET /some-path
// apm-agent-nodejs >= 2.6.0: setting default transaction name: GET unknown route
server.use('/some-path', (req, res) => {
    res.send('Hi');
});

server.listen(1212);
```

After a quick search, elastic/apm-agent-nodejs#842 looks like a possible reason.

**Comment by watson** [comment]
Thanks for letting us know about this bug and for the code so that we can easily reproduce it ❤️

I suspect this might be a regression added by #842

**Comment by watson** [comment]
This stems from how we interpret unknown routes. It's not uncommon to have a catch-all function to handle unknown routes, e.g:

```js
app.get('/foo', ...)
// ...
app.use(function (req, res) {
  res.status(404).send('Page not found')
})
```

In case an incoming request didn't match any of the above routes, it will fall down into the final route handler and a 404 will be returned. Note that this is limited to when you use `app.use(...)` and isn't an issue if you use `app.get(...)` or any equivalent HTTP method function.

First I thought we could differentiate between `app.use(fn)` and `app.use('/', fn)`, but those are handled identical inside of Express as the path defaults to `/` if it's not given.

So far I've identified the following possible solutions:

- We leave the functionality as it is, and document that it's up to the user to call [`agent.setTransactionName(...)`](https://www.elastic.co/guide/en/apm/agent/nodejs/current/agent-api.html#apm-set-transaction-name) manually in case a route is handled by an `app.use()` function.
- Treat `app.use('/', fn)` as a special case equal to how `app.use(fn)` is treated today, but treat all other paths (e.g. `app.use('/some-path', fn)`) as real paths.

I'm leaning towards the latter, though it might be weird for people that `app.use('/')` would be logged as "unknown route" whereas `app.use('/foo')` wouldn't. I'm open to other suggestions though.

As an aside, there's also the possibility to use the function name as part of the transaction name in case of an "unknown route", e.g. GET requests to the following route could be logged as `GET unknown route (catchAllRoute)`:

```js
app.use(function catchAllRoute (req, res) {
  res.status(404).send('Page not found')
})
```

**Comment by watson** [comment]
I've made a PR that implements the suggested "workaround": #1013

**Comment by MartinKolarik** [comment]
The second option is how it worked before so 👍

**Comment by watson** [comment]
This issue has now been fixed and released as part of v2.11.5. Thanks for your patience 😊
